src/coreclr/vm/threadsuspend.cpp

```diff
@@ -2577,7 +2577,13 @@ int RedirectedHandledJITCaseExceptionFilter(
     pFrame->Pop();
 
     // Copy the saved context record into the EH context;
-    ReplaceExceptionContextRecord(pExcepPtrs->ContextRecord, pCtx);
+    // NB: cannot use ReplaceExceptionContextRecord here.
+    //     these contexts may contain extended registers and may have different format
+    //     for reasons such as alignment or context compaction
+    //
+    // REVIEW: CopyContext may fail. in theory. How should we handle that? FailFast?
+    CONTEXT* pTarget = pExcepPtrs->ContextRecord;
+    CopyContext(pTarget, pTarget->ContextFlags, pCtx);
```

Comment:
It seems it would make sense to modify the ReplaceExceptionContextRecord to use CopyContext instead.

Reply:
I was not sure if `ReplaceExceptionContextRecord` can be used outside of Windows. Can it?

Reply:
I also wonder if `ReplaceExceptionContextRecord` needs to handle extended context at all if suspension resume switches completely to the `RtlRestoreContext` plan.

Reply:
> I was not sure if ReplaceExceptionContextRecord can be used outside of Windows. Can it?

I have tried moving the fix into `ReplaceExceptionContextRecord ` and that did not build on Linux, so that answer this.